src/radical/pilot/raptor/master.py

```diff
@@ -71,16 +70,18 @@ def __init__(self, cfg: ru.Config = None):
         self._reg_addr   = os.environ['RP_REGISTRY_ADDRESS']
 
         self._reg        = ru.zmq.RegistryClient(url=self._reg_addr)
+      # self._reg.dump(self._uid)
```

Comment:
btw, you don't need to create a new connection to Registry, you can read from registry after Session is initialized, thus move `self._hb_*` after line L86 and use `self._session._reg`

Reply:
and then also update this part
https://github.com/radical-cybertools/radical.pilot/blob/7bdca32ac053d691f02a9a2355222eea708643e0/src/radical/pilot/raptor/master.py#L406-L407